sys/sysent.c

```diff
@@ -0,0 +1,37 @@
+#include <sysent.h>
+#include <stdc.h>
+#include <errno.h>
+
+int sys_nosys(thread_t *td, syscall_args_t *args, int *error) {
+  log("No such system call: %ld", args->code);
+  *error = ENOSYS;
+  return -1;
+};
+
+/* This is just a stub. A full implementation of this syscall will probably
+   deserve a separate file. */
+int sys_write(thread_t *td, syscall_args_t *args, int *error) {
+  int fd = args->args[0];
+  const char *buf = (const char *)(uintptr_t)args->args[1];
+  size_t count = args->args[2];
+
+  /* TODO: copyout string from userspace */
+  if (fd == 1 || fd == 2) {
+    /* We need a local copy of the string to null-terminate it after @count
+       bytes, because the only printing facility we have is kprintf. */
+    char buf2[count + 1];
+    memcpy(buf2, buf, count);
+    kprintf("%.*s", (int)count, buf);
+
+    return count;
+  }
+
+  *error = EBADF;
```

Comment:
Oh, though in case of Linux, this behavior is only used by library function `getpriority()`, not the system call itself - which returns `20 - priority`, so that negative values are still used for errors. As we have control over both system call handlers AND library wrappers, we can always find a similar trick. No function should need entire 32-bit range for return values, even the address space is not that large.

Thus I think I'll go ahead and use negative return values for error codes.